editor.planx.uk/src/@planx/components/SetFee/utils.ts

```diff
@@ -0,0 +1,98 @@
+import { PassportFeeFields } from "@opensystemslab/planx-core/types";
+import { Store } from "pages/FlowEditor/lib/store";
+
+import { PAY_FN } from "../Pay/model";
+import { DEFAULT_PAYMENT_PROCESSING_PERCENTAGE, VAT_PERCENTAGE } from "./model";
+
+type HandleSetFees = (params: {
+  passport: Store.Passport;
+  applyCalculatedVAT: boolean;
+  fastTrackFeeAmount: number;
+  applyServiceCharge: boolean;
+  serviceChargeAmount: number;
+  applyPaymentProcessingFee: boolean;
+}) => Store.Passport["data"];
+
+// SetFee outputs any `number` type PassportFeeFields, never `boolean` ones
+type OutputFees = Omit<
+  PassportFeeFields,
+  | "application.fee.exemption.disability"
+  | "application.fee.exemption.resubmission"
+  | "application.fee.reduction.alternative"
+  | "application.fee.reduction.parishCouncil"
+  | "application.fee.reduction.sports"
+>;
+
+export const handleSetFees: HandleSetFees = ({
+  passport,
+  applyCalculatedVAT,
+  fastTrackFeeAmount,
+  applyServiceCharge,
+  serviceChargeAmount,
+  applyPaymentProcessingFee,
+}): OutputFees => {
+  // Calculated is base application fee exclusive of VAT
+  //   Any exemptions or reductions will have already set `application.fee.payable`
+  const CALCULATED_FN = "application.fee.calculated";
+  const calculated = passport.data?.[CALCULATED_FN] || 0;
+  const incomingPayable = passport.data?.[PAY_FN];
+
+  // `application.fee.payable` is total amount to be paid to GOV PAY inclusive of VAT
+  //   `application.fee.payable.VAT` is sum of all VAT-suffixed line items
+  const payable = PAY_FN;
+  const payableVAT = `${PAY_FN}.VAT`;
+
+  // At minimum, set calculated = payable if payable does not exist yet
+  const fees = {
+    [CALCULATED_FN]: calculated,
+    [payable]: incomingPayable >= 0 ? incomingPayable : calculated,
+    [payableVAT]: 0,
+  };
+
+  // Order of all extra fees and charges matters from here forwards!
+  if (applyCalculatedVAT) {
+    const calculatedVAT = calculated * VAT_PERCENTAGE;
+
+    fees[CALCULATED_FN] = calculated;
+    fees[`${CALCULATED_FN}.VAT`] = calculatedVAT;
+    fees[payable] = fees[payable] + calculatedVAT;
+    fees[payableVAT] = calculatedVAT;
+  }
+
+  const addFastTrack =
+    passport.data?.["application.fastTrack"] &&
```

Comment:
I think all of the following would meet this truthy condition -
 - `passport.data?.["application.fastTrack"]: ["false"]`
 - `passport.data?.["application.fastTrack"]: ["no"]`,
 - `passport.data?.["application.fastTrack"]: []`

Reply:
Ah perfect - thought this was an odd oversight sorry! 😅 

Might just be worth adding a comment to this effect, or to use something like `Object.hasOwn(passport.data, "application.fastTrack");` so it's clear we're checking for the presence of the key, not a truthy value.